### DISCUSSION TITLE
Auto generate comment "<!-- -->" in SSR

### DISCUSSION THREAD

**Discussion by swpaek77** [vercel/next.js#44797] [Help] [answered]
- There are very simeple code about Footer
```typescript
import React from "react";

const Footer = () => {
  return (
    <footer>
      <p>Copyright © {new Date().getFullYear()} Sample Company</p>
    </footer>
  );
};

export default Footer;
```

- Usually, we can use `new Date().getFullYear()` instead of `2023`
- Normal result is fine, but html comments are created on both sides in view source. 
![image](https://user-images.githubusercontent.com/35301536/212000800-1ba02531-1f4c-4bb3-9848-d03a7e46ff65.png)
 
- When it comes to SEO, is this the right way?
- I also found the same thing on next.js official website. 
- view-source:https://nextjs.org/
![image](https://user-images.githubusercontent.com/35301536/212001101-e5f548eb-c0e2-4913-9e17-4f65b15e8450.png)

**Comment by icyJoseph** [comment, marked as answer]
~Hi,

This is part of `react-dom`'s to string rendering. 

If you want to get rid of it do:

```jsx
const Footer = () => {
    const footerNote = `Copyright © ${new Date().getFullYear()} Sample Company`;
    return (
        <footer>
            <p>{footerNote}</p>
        </footer>
    );
};
```

Or

```jsx
const Footer = () => {
    return (
        <footer>
            <p>{`Copyright © ${new Date().getFullYear()} Sample Company`}</p>
        </footer>
    );
};
```

To see it in action:

- Go to [codesandox](https://codesandbox.io/s/new)
- Paste this into the main index file:

```jsx
import "./styles.css";

import { renderToString } from "react-dom/server";

const Footer = () => {
  return (
    <footer>
      <p>{`Copyright © ${new Date().getFullYear()} Sample Company`}</p>
    </footer>
  );
};

// const Footer = () => {
//   return (
//     <footer>
//       <p>Copyright © {new Date().getFullYear()} Sample Company</p>
//     </footer>
//   );
// };

function App() {
  return (
    <>
      <div className="App">
        <h1>Hello CodeSandbox</h1>
        <h2>Start editing to see some magic happen!</h2>
      </div>

      <Footer />
    </>
  );
}

console.log(renderToString(<App />));
```

- Look at the console
- Change which Footer renders and compare with previous step

**Reply by swpaek77** [reply]
- Thank you for quick response
- Do you think there is a problem with SEO if it made already `<!-- -->` ?

**Reply by icyJoseph** [reply]
Well, since comments are meant to be ignored by the language, mark-up language in this case, I'd say that they do not affect SEO. Lots of frameworks drop comments in your HTML. Suspense mode even drops a `<!-- $ -->` to mark boundaries.

Comments are valid HTML nodes though. They are counted as a child node, and React takes advantage of that, to mark the spot where a variable string must be placed, so called, string interpolations.

In this answer I expand a bit more as well: https://github.com/vercel/next.js/discussions/38256#discussioncomment-3070196

**Reply by swpaek77** [reply]
Thank you! It helped me a lot

**Comment by sayhicoelho** [comment]
These auto generated HTML comments are bugging the text selection. React doesn't do that. Why does Next do it?

**Reply by icyJoseph** [reply]
It's the `react-dom/server` that does right?

**Reply by sayhicoelho** [reply]
Ah, ok. It is about react-dom/server and not Next. Thanks!
